app/lib/actions/katello/host/hypervisors_update.rb

```diff
@@ -0,0 +1,62 @@
+module Actions
+  module Katello
+    module Host
+      class HypervisorsUpdate < Actions::EntryAction
+        middleware.use ::Actions::Middleware::RemoteAction
+
+        def plan(environment, content_view, hypervisor_results)
+          plan_self(:environment_id => environment.id, :content_view_id => content_view.id,
+                    :hypervisor_results => hypervisor_results)
+        end
+
+        def finalize
+          environment = ::Katello::KTEnvironment.find(input[:environment_id])
+          content_view = ::Katello::ContentView.find(input[:content_view_id])
+          hypervisor_results = input[:hypervisor_results]
+
+          %w(created updated unchanged).each do |group|
+            if hypervisor_results[group]
+              hypervisor_results[group].each do |hypervisor|
+                update_or_create_hypervisor(environment, content_view, hypervisor)
+              end
+            end
+          end
+        end
+
+        def update_or_create_hypervisor(environment, content_view, hypervisor_json)
+          name = hypervisor_json[:name]
+
+          # Since host names must be unique yet hypervisors may have unique subscription
+          # aspects in different orgs
+          host = ::Host.find_by_name(name)
+          if host
+            if host.organization.id != content_view.organization.id
```

Comment:
If the host doesn't have an org yet, this should be fatal and fail 'subscription-manager register'. (This would be equivalent to the "TODO" of differing orgs.)

The same should apply for hypervisors: No org means differing org means still TODO. I will test this case in my PR to confirm it hits the TODO.